browser/ui/views/brave_vpn/vpn_notification_dialog_view.cc

```diff
@@ -0,0 +1,142 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/brave_vpn/vpn_notification_dialog_view.h"
+
+#include <memory>
+#include <utility>
+
+#include "brave/components/brave_vpn/pref_names.h"
+#include "chrome/browser/ui/browser_tabstrip.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "components/constrained_window/constrained_window_views.h"
+#include "components/grit/brave_components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/views/controls/button/checkbox.h"
+#include "ui/views/controls/link.h"
+#include "ui/views/controls/styled_label.h"
+#include "ui/views/layout/box_layout.h"
+
+namespace brave_vpn {
+
+namespace {
+
+constexpr char kBraveVPNLearnMoreURL[] = "https://support.brave.com/";
+
+constexpr int kChildSpacing = 16;
+constexpr int kPadding = 24;
+constexpr int kTopPadding = 32;
+constexpr int kBottomPadding = 26;
+constexpr int kDialogWidth = 400;
+
+}  // namespace
+
+// static
+void VpnNotificationDialogView::Show(Browser* browser) {
+  auto* prefs = browser->profile()->GetOriginalProfile()->GetPrefs();
```

Comment:
why are we calling GetOriginalProfile here? Incognito profile prefs should be handled through the overlay pref store by adding it to `GetIncognitoPersistentPrefsAllowlist`

Reply:
it is same as window closing dialog works, took it from there, removed and using onlycurrent profile prefs